mlfinlab/portfolio_optimization/nco.py

```diff
@@ -0,0 +1,362 @@
+# pylint: disable=missing-module-docstring
+import numpy as np
+import pandas as pd
+from sklearn.covariance import LedoitWolf
+from sklearn.cluster import KMeans
+from sklearn.metrics import silhouette_samples
+from scipy.linalg import block_diag
+from mlfinlab.portfolio_optimization.risk_estimators import RiskEstimators
+
+
+class NCO:
+    """
+    This class implements the Nested Clustered Optimization (NCO) algorithm. It is reproduced with
+    modification from the following paper: `Marcos Lopez de Prado “A Robust Estimator of the Efficient Frontier”,
+    (2019). <https://papers.ssrn.com/abstract_id=3469961>`_.
+    """
+
+    def __init__(self):
+        """
+        Initialize
+        """
```

Comment:
Normally I leave a **return** statement when nothing is initialised in init(). Looks better than empty line

Reply:
Done